### PR TITLE
In py2, the signature is a unicode object

### PR DESCRIPTION
**Pull request by puiterwijk**
Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

**Comment by puiterwijk** [comment]
This really needs to be fixed before the mass updates tomorrow, as this break fedmsg-hub thoroughly.

**Comment by puiterwijk** [comment]
Okay, this was actually already fixed. We still REALLY need a new release with the basestring fix, as the version in EPEL has isinstance(..., str).

File: fedmsg/crypto/x509.py [hunk 1]
```diff
@@ -113,7 +113,8 @@ def fail(reason):
     for field in ['signature', 'certificate']:
         if field not in message:
             return fail("No %r field found." % field)
-        if not isinstance(message[field], str):
+        if not isinstance(message[field], basestring) and \
```
Comment (pypingou) [inline fedmsg/crypto/x509.py]: basestring won't work with PY3

Comment (puiterwijk) [reply fedmsg/crypto/x509.py]: basestring is the current code....

Comment (pypingou) [reply fedmsg/crypto/x509.py]: Isn't the current code using ``str``?

Comment (puiterwijk) [reply fedmsg/crypto/x509.py]: So it is, I did a terribly job rebasing. Then we *do* need this patch, just reworked.

**Comment by jeremycline** [comment]
This is a duplicate of #456, which I'll clean up and test now.

**Pull request closed without merging**